cmd/goal/node.go

```diff
@@ -98,7 +97,14 @@ func init() {
 	createCmd.Flags().StringVar(&newNodeDestination, "destination", "", "Destination path for the new node")
 	createCmd.Flags().BoolVarP(&newNodeArchival, "archival", "a", localDefaults.Archival, "Make the new node archival, storing all blocks")
 	createCmd.Flags().BoolVarP(&runUnderHost, "hosted", "H", localDefaults.RunHosted, "Configure the new node to run hosted by algoh")
-	createCmd.Flags().BoolVarP(&newNodeIndexer, "indexer", "i", localDefaults.IsIndexerActive, "Configure the new node to enable the indexer feature (implies --archival)")
+
+	// The flag for enabling an internal indexer is now deprecated, but we keep it for backwards compatibility for now.
+	var noopNodeIndexer bool
+	indexerFlagName := "indexer"
+	createCmd.Flags().BoolVarP(&noopNodeIndexer, indexerFlagName, "i", false, "")
```

Comment:
You can use `BoolP` instead of `BoolVarP` to avoid the variable:
```suggestion
	indexerFlagName := "indexer"
	_ = createCmd.Flags().BoolP(indexerFlagName, "i", false, "")
```